interactions/api/gateway/ratelimit.py

```diff
@@ -0,0 +1,77 @@
+import asyncio
+import logging
+from sys import version_info
+from time import time
+from typing import Optional
+
+log = logging.getLogger("gateway.ratelimit")
+
+
+class WSRateLimit:
+    """
+    A class that controls Gateway ratelimits using locking and a timer.
+
+    .. note ::
+        While the docs state that the Gateway ratelimits are 120/60 (120 requests per 60 seconds),
+        this ratelimit offsets to 115 instead of 120 for room.
+
+    :ivar Lock lock: The gateway Lock object.
+    :ivar int max: The upper limit of the ratelimit. Defaults to `115` seconds.
```

Comment:
```suggestion
    :ivar int max: The upper limit of the ratelimit. Defaults to `115` requests per second.
```
Should be this, right?

Reply:
`max` is in seconds, having it be requests for second doesn't do much I guess?

Reply:
When the limit is `115` per 60 seconds why is the max then `115` seconds? This isn't clear to me

Reply:
Because the "per 60" seconds part is defined in the `per_second` attribute

Reply:
Ok but this still doesn't explain why max is in seconds?

Reply:
yeah it does?
"upper limit of the ratelimit, default value's in seconds"
I don't see what you're getting at but I want to understand your perspective

Reply:
Honestly idk myself, nvm ig